.github/workflows/tf-plan.yml

```diff
@@ -0,0 +1,74 @@
+# "on" step - only in tf folder, on all pr
+# git checkout 
+# set up tf
+# gcloud authenticate - use secrets
+# tf init
+#tf plan 
+# output plan as comment
+# fail action if plan failed
+name: Terraform Plan
+
+on:
+  pull_request:
+    paths:
+      - 'tf/**'
+
+jobs:
+  terraform:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+      
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v1'
+
+    - name: Authenticate with GCP
+      uses: 'google-github-actions/auth@v1'
+      with:
+        credentials_json: ${{ secrets.GCP_SA_KEY }}
+
+    - name: Set up Terraform
+      uses: hashicorp/setup-terraform@v2 
+      
+    - run: terraform init
+
+    - id: plan
+      run: terraform plan -no-color
```

Comment:
You need to pass to this command the location of the tfvars file i think you do it with -f (I'll check later when I'm near my pc)